### PR TITLE
added es6 migrations to configuration.js and db.js

### PR DESCRIPTION
**Pull request by mandeepm91**
refs #9589

Applied ES6 migrations to files configuration.js and db.js in the server/api folder

### CODE REVIEW

File: core/server/api/db.js [hunk 1]
```diff
@@ -120,21 +126,21 @@ db = {
...
         function deleteContent() {
-            return models.Base.transaction(function (transacting) {
+            return models.Base.transaction((transacting) => {
                 queryOpts.transacting = transacting;
 
                 return models.Post.findAll(queryOpts)
                     .then((response) => {
                         return Promise.map(response.models, (post) => {
-                            return models.Post.destroy(_.merge({id: post.id}, queryOpts));
+                            return models.Post.destroy(merge({id: post.id}, queryOpts));
```
Comment (vikaspotluri123) [inline core/server/api/db.js]: //cc @kirrg001 can you use `Object.assign` here?

Comment (mandeepm91) [reply core/server/api/db.js]: @vikaspotluri123 Thanks. we can do this. I am updating the PR

File: core/server/api/configuration.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ function getAboutConfig() {
         version: ghostVersion.full,
         environment: config.get('env'),
         database: config.get('database').client,
-        mail: _.isObject(config.get('mail')) ? config.get('mail').transport : ''
+        mail: Object(config.get('mail')) === config.get('mail') ? config.get('mail').transport : ''
```
Comment (kirrg001) [inline core/server/api/configuration.js]: Hmm this replacement is also true when config.get('mail') is an Array. But also `isObject` was detecting that.
I think what we need here is `isPlainObject`.

**Review by kirrg001 — CHANGES_REQUESTED** [review]
Just left one comment. Otherwise LGTM 👍 
Thanks :)

**Comment by mandeepm91** [comment]
Thanks for the review @kirrg001 , I've updated the PR.

**Review by kirrg001 — APPROVED** [review]

**Pull request merged**